pkg/datapath/loader/hash.go

```diff
@@ -29,6 +30,23 @@ func (d datapathHash) hashEndpoint(c datapath.ConfigWriter, nodeCfg *datapath.Lo
 	if err := c.WriteEndpointConfig(h, nodeCfg, epCfg); err != nil {
 		return "", err
 	}
+
+	// Include endpoint configuration in the hash, otherwise different runtime
+	// configurations will hash to the same value and the update will be skipped.
+	if epCfg.IsHost() {
+		cfg, _ := ciliumHostRewrites(epCfg)
+		_, err := fmt.Fprintf(h, "%+v", cfg)
+		if err != nil {
+			return "", err
```

Comment:
nit: this err bubbled pretty far up the stack and it's from a Fprintf call, might cause for some confusing errors one day without some error context?